components/brave_wallet_ui/components/extension/transaction-box/index.tsx

```diff
@@ -9,6 +9,15 @@ export interface Props {
   transactionInfo: TransactionInfo
 }
 
+const mapTransactionTypeToText = {
+  [TransactionType.ETHSend]: 'ETHSend',
+  [TransactionType.ERC20Transfer]: 'ERC20Transfer',
+  [TransactionType.ERC20Approve]: 'ERC20Approve',
+  [TransactionType.ERC721TransferFrom]: 'ERC721TransferFrom',
+  [TransactionType.ERC721SafeTransferFrom]: 'ERC721SafeTransferFrom',
+  [TransactionType.Other]: 'Other'
```

Comment:
Would something like this work so we don't need to maintain a list over time?
https://stackoverflow.com/a/42172322

Reply:
@bbondy I think that was what we original using? Seems no longer working with the generated type.

Reply:
An easy way we could achieve this would be to  use `Object.keys()`

So you could do `const txKeys = Object.keys(TransactionType)`

and then `<DetailText>{txKeys[txType]}</DetailText>`

Reply:
https://dmitripavlutin.com/access-object-keys-values-entries/#1-objectkeys-returns-keys

Reply:
Updated, thanks a lot @Douglashdaniel !